dev/clint/src/clint/rules.py

```diff
@@ -11,12 +13,18 @@
 
 class Rule(ABC):
     _CLASS_NAME_TO_RULE_NAME_REGEX = re.compile(r"(?<!^)(?=[A-Z])")
+    _id_counter = itertools.count(start=1)
+    _generated_id: str
+
+    def __init_subclass__(cls, **kwargs):
+        super().__init_subclass__(**kwargs)
+        # Only generate ID for concrete classes
+        if not inspect.isabstract(cls):
+            cls._generated_id = f"MLF{next(cls._id_counter):04d}"
 
-    @abstractmethod
     def _id(self) -> str:
```

Comment:
let's remove this method and make the `id` property return `self._generated_id`.